### DISCUSSION TITLE
Card Negative Watts vs Postive Watts

### DISCUSSION THREAD

**Discussion by Roving-Ronin** [slipx06/sunsynk-power-flow-card#463] [General]
Hi @slipx06 

Dumb question, but just wondering how come some sections of the card show a negative watts (i.e. -900W) when it is drawing/consuming power, however other parts that are consuming power are shown as a positive value normally?

Just looking at the examples on the Repos readme/front page, or even the 'original' screenshots off a SunSynk inverter, as you posted for the flow logic at: https://github.com/slipx06/sunsynk-power-flow-card/discussions/280#discussioncomment-8657562 and they all do this.

Wondering if the Battery for instance shows '900W' when its discharging (that makes sense as its 'providing not consuming') and then conversely shows '-900W' when its charging (as its consuming the power at that point) and the Grid box is the same showing a positive number when pulling power from the grid and a negative when exporting (i.e the grid is consuming it)....

Then how come the other power users like the Essential and Non-Essential Power are shown consuming power, but always as a positive number of watts, instead of a negative?

:-)

**Comment by slipx06** [comment]
Ideally you don't want to see a negative number but rather use the animation to determine the direction of power flow. The negative or positive sensor values shown when consuming/producing were based on the Modbus registers values on the Sunsynk inverter. Its true that there is some inconsistency and this is also true for the different inverter models. The invert_ option was introduced to handle this and also the show_absolute setting for battery, aux and grid.

**Comment by Breinholm-coder** [comment]
@slipx06 "Yes, that's true - In the physical sense, power—being the rate of energy transfer—is always positive. However, this isn't accurate when using math to calculate incoming or outgoing power flows from units (boxes in your flow chart). I prefer to follow the current directions in my calculations, where power flowing into my battery is positive and power flowing out is negative. If I do this, I can choose between having my negative power displayed correctly, but the flow indicators (dots) moving in the wrong direction, or having the flow indicators (dots) moving correctly but displaying the wrong power sign. Your 'Invert Power' option doesn't solve this! 
An 'invert flow direction' flag would be very helpful, allowing people to have flows moving towards negative power values if they prefer this notation."

**Reply by Breinholm-coder** [reply]
"I just want to add that I'm considering having one flag per flow path, which would make the preferred flow direction completely free to choose."

**Reply by slipx06** [reply]
Its a good idea and not too hard to implement

**Reply by slipx06** [reply]
@Breinholm-coder added this option to https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v5.3.0

**Reply by Breinholm-coder** [reply]
@slipx06 
Excellent! - You are quick👍 
Thanks / Breinholm

**Reply by vinceg0267** [reply]
There is a bug that should be fixed in this release: if in battery options, invert power is false and invert flow is true, when the battery is charging the indicator shows the time remaining to the end of the discharge (20%) instead of showing the time remaining to full charge (100%)

**Reply by slipx06** [reply]
I've updated the logic. You can try https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v5.3.2

**Reply by vinceg0267** [reply]
@slipx06 
Well done! You're great!
